README.md

```diff
@@ -1,8 +1,18 @@
 [![Build and Test](https://github.com/dafny-lang/dafny/workflows/Build%20and%20Test/badge.svg)](https://github.com/dafny-lang/dafny/actions?query=workflow%3A%22Build+and+Test%22) [![Gitter](https://badges.gitter.im/dafny-lang/community.svg)](https://gitter.im/dafny-lang/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
 
+Dafny is a **programming language teamed up with a verifier**. As you type in your program, the verifier constantly looks over your shoulder, flags any errors, shows you counterexamples, and congratulates you when your code matches your specifications. When you're done, Dafny can **compile your code to C#, Java, JavaScript or Go**, so it can integrate with your existing workflow.
+
+![vs-code-dafny-2 0 0-demo](https://user-images.githubusercontent.com/3601079/140799975-f3ac0925-10d9-4c14-b1a9-cd449854c6ae.gif)
```

Comment:
Absolutely love having the animated gif here. We should re-record it to reflect the latest UX though since it's been improved substantially.

I'm also not sure we want to show the counterexample feature just yet, since it looks nice for this super-simple example but needs some love to scale to realistically complex code.

Reply:
Where do you think the UI has been improved substantially? I can note that there might be a temporary "Verifying Test..." that would be barely visible, but besides that, there is no difference with the current UI, right? Maybe the speed of verification?

I think it's fine to keep the counter-example feature there for the purpose of the demo, because it's currently the only way to illustrate that Dafny will be able to provide useful feedback in the future. It has been here for a long time, and it's still useful to some extend. But I agree it needs some love.

Reply:
Alright, I'd definitely rather have a very slightly out of date gif than no gif, so let's go with this version!